src/com/unbounce/treajure/monad.clj

```diff
@@ -0,0 +1,34 @@
+(ns com.unbounce.treajure.monad
+  (:refer-clojure :hide [do])
+  (:require [monads.core :refer [>>=] :as monads])
+  (:import [monads.types Bind Return]))
+
+(defmulti monadic-value? type)
+
+(defmethod monadic-value? Bind [_]
+  true)
+
+(defmethod monadic-value? Return [_]
+  true)
+
+(defmethod monadic-value? :default [_]
+  false)
+
+(defmacro do [& instr]
+  `(monads/mdo ~@instr))
+
+(def return monads/return)
+
+(defn bind-ignore
+  "Executes monadic sub-routine withhout binding it's return value"
```

Comment:
Would be good to expand this comment a bit with an example of why this is useful.